connectors/sources/sharepoint.py

```diff
@@ -570,7 +574,7 @@ async def get_list_items(self, list_id, site_url, server_relative_url, **kwargs)
         ):
             for result in list_items_data:
                 if not result.get("Attachments"):
-                    url = f"{self.host_url}{server_relative_url}/DispForm.aspx?ID={result['Id']}&Source={self.host_url}{server_relative_url}/AllItems.aspx&ContentTypeId={result['ContentTypeId']}"
+                    url = f"{self.host_url}{quote(server_relative_url)}/DispForm.aspx?ID={result['Id']}&Source={self.host_url}{quote(server_relative_url)}/AllItems.aspx&ContentTypeId={result['ContentTypeId']}"
```

Comment:
can we add in the class a dedicated function for building URLs, I see now places where you use `join` and places where you use string formatting.

Ideally we have one single function that gets all parts and do the quoting work